### PR TITLE
Filebeat refactoring and cleanup

### PR DESCRIPTION
**Pull request by ruflin**
* Cleanup variable naming
* Update state also if lines are skipped / excluded
* Introduce option to update state without sending an event

### CODE REVIEW

File: filebeat/crawler/prospector_log.go [hunk 1]
```diff
@@ -265,19 +265,20 @@ func (p *ProspectorLog) continueExistingFile(h *harvester.Harvester, previousFil
 func (p *ProspectorLog) resumeHarvesting(h *harvester.Harvester, offset int64) {
 
 	logp.Debug("prospector", "Start / resuming harvester of file: %s", h.Path)
-	h.SetOffset(offset)
-	h.Start()
+	newHarvester, _ := p.Prospector.AddHarvester(h.Path, h.Stat)
+	newHarvester.SetOffset(offset)
+	newHarvester.Start()
```
Comment (urso) [inline filebeat/crawler/prospector_log.go]: why creating a new harvester here?

We need a harvester.Copy/Clone method?

File: filebeat/harvester/log.go [hunk 2]
```diff
@@ -132,29 +88,45 @@ func (h *Harvester) Harvest() {
...
+			event.ReadTime = ts
+			event.Bytes = bytesRead
+			event.Text = &text
+			event.JSONFields = jsonFields
 		}
 
-		// Set Offset
-		h.SetOffset(h.GetOffset() + int64(bytesRead)) // Update offset if complete line has been processed
+		// Always send event to update state, also if lines was skipped
+		h.sendEvent(event)
```
Comment (urso) [inline filebeat/harvester/log.go]: In order to update registrar?

Comment (ruflin) [inline filebeat/harvester/log.go]: Yes

File: filebeat/crawler/prospector_log.go (hunk 1 above)
Comment (ruflin) [inline filebeat/crawler/prospector_log.go]: @urso I removed this code again as it is not needed. I was playing around with creating a new harvester every time offset changes etc. And this is a left over. Potentially we need something like harvester copy.

File: filebeat/harvester/log.go [hunk 3]
```diff
@@ -132,29 +88,45 @@ func (h *Harvester) Harvest() {
...
+		Offset:        h.GetOffset(),
+		Bytes:         0,
+		Fileinfo:      &h.fileInfo,
+		JSONConfig:    h.Config.JSON,
+	}
+}
+
+// sendEvent sends event to the spooler channel
+func (h *Harvester) sendEvent(event *input.FileEvent) {
+	h.SpoolerChan <- event // ship the new event downstream
```
Comment (urso) [inline filebeat/harvester/log.go]: what if spooler is not reading from channel due to being blocked in libbeat output? We need to 'unblock' here?

Comment (ruflin) [inline filebeat/harvester/log.go]: @urso Agree. The behaviour in this PR did not change from previous behaviour. This kind of changes should go into https://github.com/elastic/beats/pull/964

File: filebeat/harvester/log.go [hunk 4]
```diff
@@ -132,29 +88,45 @@ func (h *Harvester) Harvest() {
 			return
 		}
 
+		// Update offset if complete line has been processed
+		h.SetOffset(h.GetOffset() + int64(bytesRead))
```
Comment (urso) [inline filebeat/harvester/log.go]: h.Inc/AddOffset

**Comment by urso** [comment]
LGTM

**Pull request merged**